tools/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala

```diff
@@ -94,6 +97,49 @@ class SQLPlanParserSuite extends FunSuite with BeforeAndAfterEach with Logging {
     }
   }
 
+  test("Error parser does not cause entire app to fail") {
+    // The purpose of this test is to make sure that the SQLParser won't trigger an exception that
+    // causes the entire app analysis to fail.
+    // In order to simulate unexpected scenarios, the eventlog sqlparse_failure is injected with
+    // faulty expressions.
+    // For example:
+    //    Filter (((value#8 <> 100) AND (value#8 > 50)) OR (value#8 = 0))
+    //    One of the predicates is converted from "<" to "<>".
+    //    The latter is an invalid predicate causing the parser to throw a scala-match error.
```

Comment:
Can we do this corruption on the fly in a temp directory with some existing event log without adding another 82+KB binary resource ?

Reply:
Given the priorities of the task, I don't have a quick solution to inject faulty events in this PR.
I agree that we need something better.
One way is to extend `ToolTestUtils.generateEventLog` to give more control on the generated eventlog.
The other way which I prefer not to do is to alter the eventlog as a text file.

Reply:
Removed the eventlog files and used on-the-fly injection